ibis/expr/types/relations.py

```diff
@@ -263,62 +283,123 @@ def _ensure_expr(self, expr):
+        >>> t.columns
+        ['name',
+         'height',
+         'mass',
+         'hair_color',
+         'skin_color',
+         'eye_color',
+         'birth_year',
+         'sex',
+         'gender',
+         'homeworld',
+         'species',
+         'films',
+         'vehicles',
+         'starships']
+        """
+        return list(self.schema().names)
 
     def schema(self) -> sch.Schema:
-        """Get the schema for this table (if one is known).
+        """Return the schema for this table.
 
         Returns
         -------
         Schema
             The table's schema.
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.examples.starwars.fetch()
+        >>> t.schema()
+        ibis.Schema {
+          name        string
+          height      int64
+          mass        float64
+          hair_color  string
+          skin_color  string
+          eye_color   string
+          birth_year  float64
+          sex         string
+          gender      string
+          homeworld   string
+          species     string
+          films       string
+          vehicles    string
+          starships   string
+        }
         """
         return self.op().schema
 
-    def group_by(self, by=None, **group_exprs: Any) -> GroupedTable:
+    def group_by(
+        self,
+        by: str | ir.Value | Iterable[str] | Iterable[ir.Value] | None = None,
+        **key_exprs: str | ir.Value | Iterable[str] | Iterable[ir.Value],
+    ) -> GroupedTable:
         """Create a grouped table expression.
 
         Parameters
         ----------
         by
             Grouping expressions
-        group_exprs
+        key_exprs
             Named grouping expressions
 
-        Examples
-        --------
-        >>> import ibis
-        >>> from ibis import _
-        >>> t = ibis.table(dict(a='int32', b='timestamp', c='double'), name='t')
-        >>> t.group_by([_.a, _.b]).aggregate(sum_of_c=_.c.sum())
-        r0 := UnboundTable: t
-          a int32
-          b timestamp
-          c float64
-        Aggregation[r0]
-          metrics:
-            sum_of_c: Sum(r0.c)
-          by:
-            a: r0.a
-            b: r0.b
-
         Returns
         -------
         GroupedTable
             A grouped table expression
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable({"fruit": ["apple", "apple", "banana", "orange"], "price": [0.5, 0.5, 0.25, 0.33]})
+        >>> t
+        ┏━━━━━━━━┳━━━━━━━━━┓
+        ┃ fruit  ┃ price   ┃
+        ┡━━━━━━━━╇━━━━━━━━━┩
+        │ string │ float64 │
+        ├────────┼─────────┤
+        │ apple  │    0.50 │
+        │ apple  │    0.50 │
+        │ banana │    0.25 │
+        │ orange │    0.33 │
+        └────────┴─────────┘
+        >>> t.group_by("fruit").agg(total_cost=_.price.sum(), avg_cost=_.price.mean())
```

Comment:
Another errant `_`

Reply:
This all turns out to be because once you import something it stays in scope for the entire pytest session 😬

Reply:
:grimacing: that is... not great